RecoParticleFlow/PFClusterProducer/src/SpikeAndDoubleSpikeCleaner.cc

```diff
@@ -117,10 +117,55 @@ clean(const edm::Handle<reco::PFRecHitCollection>& input,
     }    
     const spike_cleaning& clean = _thresholds.find(hitlayer)->second;    
     if( rechit.energy() < clean._singleSpikeThresh ) continue;
+
+
+
+    //Fix needed for HF
+    float compsumE = 0.0;
+    if ((hitlayer==12 or hitlayer==11))
+	{
+	int predphi =2;
+
+	int comp = std::abs(hitlayer-13);
+	const HcalDetId& detid = (HcalDetId)rechit.detId();
+	std::vector<uint32_t> RawDetIds;
+	int heta = detid.ieta();
+	int hphi = detid.iphi();
+	
+	if (std::abs(heta)>39) predphi=4;
+
+ 	int curphiL = hphi-predphi;
+	int curphiH = hphi+predphi;
+
+	if (predphi>abs(hphi) or predphi>abs(72-hphi))
```

Comment:
Yes, let it hardcoded. But please extract it from the "HF valid phi numbering range", whenever it is stored/defined. So that if it gets ever modified you don't have to modify several hidden lines of code here and there.